src/components/ViewComments.jsx

```diff
@@ -1,26 +1,90 @@
-const ViewComments = ({ isLoadingComments, isReadingComments, comments }) => {
+import { useEffect, useState } from "react";
+import { postCommentToReview } from "../api";
+import { getCommentsOnReview } from "../api";
+
+const ViewComments = ({ isReadingComments, review_id }) => {
+  const [comment, setComment] = useState("");
+  const [comments, setComments] = useState([]);
+  const [isLoadingComments, setIsLoadingComments] = useState(false);
+  const [postError, setPostError] = useState(false);
+
+  useEffect(() => {
+    setIsLoadingComments(true);
+    getCommentsOnReview(review_id).then((data) => {
+      setComments(data.reverse());
+      setIsLoadingComments(false);
+    });
+  }, []);
+
+  const handleSubmit = (event, review_id) => {
```

Comment:
At the minute I can post an empty comment - it would be good to have some logic to prevent this.

On a slow connection I don't know if my comment has been posted until the request comes back, depending on how long this takes I might click the button lots of times, It would be good to add in some optimistic rendering so that I instantly see my comments been posted.

Reply:
i couldnt figure this out earlier but i have a working method now :)